spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/RuleInterpreter.scala

```diff
@@ -76,6 +80,46 @@ trait RuleInterpreter {
     (intTry orElse longTry orElse doubleTry orElse boolTry) getOrElse lit(input)
   }
 
+  /**
+   * Preforms a simple type cast to input base on the DataType
+   * @param input Value to be casted
+   * @param dataType DataType of the value to be casted to
+   * @return Returns Column representation of the newly casted value
+   */
+  def simpleLiteralCast(input: String, dataType: DataType): Column = {
+    Try({
```

Comment:
I would make this `Try()` into a function returning `Try[Column]`. Then it can be used in `inferStrictestType` too, by smartly cyclining through the types and we got consistent behavior from both `inferStrictestType` and `simpleLiteralCast`

Reply:
I am not sure I get what you mean

Reply:
It's just a suggestion, how to make `lit` behave similarly (consistently) as `fillNulls`. Doesn't affect you solution, that does exactly what it needs.
1) Makte the `Try(...)` expression into a private function
2) Call the function in your new `simpleLiteralCast`
3) In `inferStrictestType` in a cycle of calls to this function with different datatypes and `Try`'s recover find the strictest type applicable.

Bonus for 3* - it can be done smart, if it fails on Double, no point to try on Float etc.

Reply:
I don't want to rewrite lit here and broaden the scope of this hot-feature. But I think I get what you mean.

Reply:
Fair enough, you are right.
That's why I am not fond of hot features.